runtime/server_http_request_test.go

```diff
@@ -737,3 +739,45 @@ func TestGetQueryValues(t *testing.T) {
 	assert.Equal(t, []string(nil), lastQueryParam)
 	assert.Equal(t, 0, len(lastQueryParam))
 }
+
+func TestPeekBody(t *testing.T) {
+	gateway, err := benchGateway.CreateGateway(
+		defaultTestConfig,
+		defaultTestOptions,
+		exampleGateway.CreateGateway,
+	)
+
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer gateway.Close()
+
+	bgateway := gateway.(*benchGateway.BenchGateway)
+	bgateway.ActualGateway.HTTPRouter.Register(
+		"POST", "/foo", zanzibar.NewRouterEndpoint(
+			bgateway.ActualGateway.Logger,
+			bgateway.ActualGateway.AllHostScope,
+			"foo", "foo",
+			func(
+				ctx context.Context,
+				req *zanzibar.ServerHTTPRequest,
+				res *zanzibar.ServerHTTPResponse,
+			) {
+				rawBody, success := req.ReadAll()
+				assert.True(t, success)
+				req.RawBody = rawBody
```

Comment:
this breaks encapsulation. `RawBody` should be private not public. `ReadAll()` should assign to it.

Can you refactor `RawBody` => `rawBody`